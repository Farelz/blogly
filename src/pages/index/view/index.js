import React from 'react';
import types from 'prop-types';
import PostCard from './post-card';
import { sort_posts } from '../../../lib/helpers';

export class Body extends React.Component {
    componentDidMount() {
        window.addEventListener('scroll', this.onScroll, false);
    }
    componentWillUnmount() {
        window.removeEventListener('scroll', this.onScroll, false);
    }
    onScroll = () => {
        if (window.innerHeight + window.scrollY >= document.body.offsetHeight) {
            this.props.fetch_more();
        }
    };

    render_post_cards = (posts) => {
        if (posts.length > 0) {
            return posts.map((obj) => (
                <div key={obj.node.id} className="col l6 m10 s12">
                    <PostCard post={obj.node} />
                </div>
            ));
        } else {
            return (
                <div className="col s12">
                    <h5 className="center">They{'\''}re currently no post available.</h5>
                </div>
            );
        }
    };
    render() {
        const posts = sort_posts(this.props.posts);

        return (
            <div className="body section container">
                <div className="row">{this.render_post_cards(posts)}</div>
            </div>
        );
    }
}

Body.propTypes = {
    posts: types.array.isRequired,
    fetch_more: types.func.isRequired
};

export default Body;
