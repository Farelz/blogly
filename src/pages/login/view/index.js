import React from 'react';
import types from 'prop-types';
import cookie from 'cookie';
import Link from 'next/link';
import { Mutation, withApollo } from 'react-apollo';
import { LoginForm } from '../../../components/forms';
import { tokenKey } from '../../../keys';
import redirect from '../../../lib/redirect';
import mutation from './query';

export const PageView = ({ client }) => {
    const onCompleted = (data) => {
        document.cookie = cookie.serialize(tokenKey, data.loginUser.token, {
            maxAge: 30 * 24 * 60 * 60 // 30 days
        });

        client.resetStore().then(() => redirect({}, '/'));
    };

    const handleSubmit = (e, loginUser) => {
        e.preventDefault();
        e.stopPropagation();
        let email = e.target['email'];
        let password = e.target['password'];

        loginUser({
            variables: {
                email: email.value,
                password: password.value
            }
        });

        email.value = password.value = '';
    };

    return (
        <Mutation mutation={mutation} onCompleted={onCompleted}>
            {(loginUser, { loading, error }) => {
                const err =
                    error &&
                    error
                        .toString()
                        .split(':')
                        .pop();

                return (
                    <div className="auth auth-login">
                        <div className="container">
                            <div className="login-section section center z-depth-1">
                                <div className="heading">
                                    <h5>Welcome Back</h5>
                                    <p>
                                        Sign in to share your story with the world, appreciate
                                        stories you love, and more.
                                    </p>
                                </div>

                                <div className="login-form">
                                    <LoginForm
                                        loggingIn={loading}
                                        auth_message={err}
                                        onSubmit={(e) => handleSubmit(e, loginUser)}
                                    />
                                </div>

                                <div className="extra-info">
                                    <p>
                                        Don{'\''}t have an account?
                                        <Link href="/signup">
                                            <a> Signup</a>
                                        </Link>
                                    </p>
                                </div>
                            </div>
                        </div>
                    </div>
                );
            }}
        </Mutation>
    );
};

PageView.propTypes = {
    client: types.object
};

export default withApollo(PageView);
