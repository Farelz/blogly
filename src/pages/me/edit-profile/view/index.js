/**
 * ./src/components/admin/edit-profile/view/body
 */

import React, { Component } from 'react';
import types from 'prop-types';

// import { createToast } from '../../../../lib/helpers';

import { EditPicture } from './editPicture';
import { EditDetails } from './editDetails';

export class Body extends Component {
    constructor(props) {
        super(props);

        this.state = this.getInitStateFromProps(props) || {
            display_name: '',
            description: '',
            pic_url: ''
        };
    }
    getInitStateFromProps = (props) => {
        if (props.data) {
            const { fullName, description, gravatarUrl } = props.data;

            return {
                display_name: fullName,
                description,
                pic_url: gravatarUrl
            };
        }
    };

    onChange = (e) => {
        e.preventDefault();
        this.setState({
            [e.target.id]: e.target.value
        });
    };
    afterUpload = (pic_url) => {
        this.setState({
            pic_url
        });
    };

    sortNewData = (data) => {
        let user_data = {};

        if (data !== undefined) {
            const { fullName, description } = data;
            const { display_name: newName, description: newDesc } = this.state;

            if (newName !== fullName) {
                user_data = {
                    ...user_data,
                    newName
                };
            }

            if (newDesc !== description) {
                user_data = {
                    ...user_data,
                    newDesc
                };
            }
        }

        return user_data;
    };

    onSaveClick = (e) => {
        e.preventDefault();

        if (this.props.data) {
            const user_data = this.sortNewData(this.props.data);
            this.props.handleNewProfile(user_data);
        }
    };

    render() {
        const { display_name, description, pic_url } = this.state;
        return (
            <div className="edit-profile">
                <div className="container">
                    <div className="row">
                        <div className="col s12 m5 push-m7">
                            <EditPicture init_image={pic_url} afterUpload={this.afterUpload} />
                        </div>
                        <div className="col s12 m7 pull-m5">
                            <EditDetails
                                init_data={{ display_name, description }}
                                onChange={this.onChange}
                            />
                        </div>
                    </div>
                    <button className="btn" onClick={this.onSaveClick}>
                        Save
                    </button>
                </div>
            </div>
        );
    }
}

Body.propTypes = {
    data: types.object.isRequired,
    handleNewProfile: types.func.isRequired
};

export default Body;
