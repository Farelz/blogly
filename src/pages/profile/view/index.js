import React from 'react';
import types from 'prop-types';
import withTopbar from '../../../components/app';
import Error from '../../../components/error';

import AuthorInfo from './author-info';
import UserPosts from './user-posts';

export class PageView extends React.Component {
    render() {
        if (this.props.user === undefined) {
            return (
                <Error
                    render={
                        <div>
                            <h5 className="center">User not found</h5>
                        </div>
                    }
                />
            );
        } else {
            return (
                <div className="page-body section container">
                    <div className="row">
                        <div className="col s12">
                            <div>
                                <AuthorInfo data={this.props.user} />
                                <UserPosts data={this.props.user} />
                            </div>
                        </div>
                    </div>
                </div>
            );
        }
    }
}

PageView.propTypes = {
    user: types.object
};

export default withTopbar(PageView);
