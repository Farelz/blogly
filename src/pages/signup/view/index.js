import React from 'react';
import cookie from 'cookie';
import types from 'prop-types';
import Link from 'next/link';
import { Mutation, withApollo } from 'react-apollo';

import { SignupForm } from '../../../components/forms';
import { tokenKey } from '../../../keys';
import redirect from '../../../lib/redirect';
import mutation from './query';

export const PageView = ({ client }) => {
    const onCompleted = (data) => {
        document.cookie = cookie.serialize(tokenKey, data.createUser.token, {
            maxAge: 30 * 24 * 60 * 60 // 30 days
        });

        client.resetStore().then(() => redirect({}, '/'));
    };

    const handleSubmit = (e, func) => {
        e.preventDefault();
        e.stopPropagation();
        let email = e.target['email'];
        let password = e.target['password'];
        let username = e.target['username'];
        let fullName = e.target['full_name'];

        func({
            variables: {
                email: email.value,
                password: password.value,
                username: username.value,
                fullName: fullName.value
            }
        });

        // email.value = password.value = username.value = fullName.value = '';
    };
    return (
        <Mutation mutation={mutation} onCompleted={onCompleted}>
            {(createUser, { loading, error }) => {
                const err =
                    error &&
                    error
                        .toString()
                        .split(':')
                        .pop();

                return (
                    <div className="auth auth-signup">
                        <div className="container">
                            <div className="signup-section section center z-depth-1">
                                <div className="heading">
                                    <h5>Join ReactPress</h5>
                                    <p>
                                        Sign up to share your story with the world, appreciate
                                        stories you love, and more.
                                    </p>
                                    <p>
                                        * Password must contain atleast one uppercase letter or one
                                        number. And must be atleast 6 characters long
                                    </p>
                                </div>

                                <div className="login-form">
                                    <SignupForm
                                        loggingIn={loading}
                                        auth_message={err}
                                        onSubmit={(e) => handleSubmit(e, createUser)}
                                    />
                                </div>

                                <div className="extra-info">
                                    <p>
                                        Have an account?
                                        <Link href="/login">
                                            <a> Login</a>
                                        </Link>
                                    </p>
                                </div>
                            </div>
                        </div>
                    </div>
                );
            }}
        </Mutation>
    );
};

PageView.propTypes = {
    client: types.object
};

export default withApollo(PageView);
