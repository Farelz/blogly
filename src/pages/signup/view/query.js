import gql from 'graphql-tag';

export default gql`
    mutation createUser(
        $fullName: String!
        $email: String!
        $password: String!
        $username: String!
    ) {
        createUser(fullName: $fullName, email: $email, password: $password, username: $username) {
            token
        }
    }
`;
