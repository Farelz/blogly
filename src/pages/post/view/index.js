import React from 'react';
import types from 'prop-types';
import Router from 'next/router';
import { Query } from 'react-apollo';

import Error from '../../../components/error';
import query from './query';
import PostBody from './body';
// import { createToast } from '../../../lib/helpers';

export class PageBody extends React.Component {
    // event handlers
    handleClap = async () => {
        // if (this.props.loggedIn === true) {
        //     const post_id = this.props.post.node.uuid;
        //     const clap_data = {
        //         post_id
        //     };
        //     try {
        //         await this.props.clap(clap_data);
        //     } catch (e) {
        //         createToast('There was an error, please try again');
        //     }
        // } else {
        //     createToast('Sign up or Login to appreciate this post');
        // }
    };
    handleComment = async (body) => {
        return body;
        // if (this.props.loggedIn === true) {
        //     const post_id = this.props.post.node.uuid;
        //     const comment_data = {
        //         body,
        //         post_id
        //     };

        //     try {
        //         await this.props.comment(comment_data);
        //     } catch (e) {
        //         createToast('There was an error, please try again');
        //     }
        // } else {
        //     createToast('Sign up or Login to comment on this post');
        // }
    };
    hadleCommentReply = async (body, parent_id) => {
        return body, parent_id;
        // if (this.props.loggedIn === true) {
        //     const post_id = this.props.post.node.uuid;
        //     const reply_data = {
        //         body,
        //         parent_id,
        //         post_id
        //     };

        //     try {
        //         await this.props.reply_comment(reply_data);
        //     } catch (e) {
        //         createToast('There was an error, please try again');
        //     }
        // } else {
        //     createToast('Sign up or Login to reply to this comment');
        // }
    };

    // render method
    renderError = () => {
        return (
            <Error
                render={
                    <div className="center">
                        <div className="center-align">
                            <img className="responsive-img" src="/static/404.png" />
                        </div>
                        <h5 className="center">
                            The content you{'\''}re looking for is currently not available
                        </h5>
                        <a onClick={() => Router.back()}>Go back to the previous page</a>
                    </div>
                }
            />
        );
    };
    render() {
        const { post_name } = this.props;
        return (
            <Query query={query} variables={{ post_name }}>
                {({ error, data}) => {
                    // if there's an error
                    if (error) return <Error render={<p>Error fetching post</p>} />;
                    
                    return data.post !== null ? <PostBody
                        post={data.post}
                        onClap={this.handleClap}
                        onComment={this.handleComment}
                        onCommentReply={this.hadleCommentReply}
                    />: this.renderError();
                }}
            </Query>
        );
    }
}

PageBody.propTypes = {
    post_name: types.string.isRequired,
    loggedIn: types.bool.isRequired
};

export default PageBody;
