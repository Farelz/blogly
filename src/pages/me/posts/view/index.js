import React from 'react';
import types from 'prop-types';
import Link from 'next/link';
import { Query, Mutation, withApollo } from 'react-apollo';

import Error from '../../../../components/error';
import { createToast } from '../../../../lib/helpers';

import query, { deleteMutation } from './query';
import { Post } from './post';

const PageView = ({ client }) => {
    const render_page_header = () => {
        return (
            <div className="posts-info">
                <h1>Your posts</h1>
                <Link href={'/me/new-post'}>
                    <a className="btn btn-flat">New Post</a>
                </Link>
            </div>
        );
    };

    const render_no_post = () => {
        return (
            <div className="container">
                <h5 className="center">No Posts</h5>
            </div>
        );
    };

    const render_posts = (posts, delete_post) => {
        return posts.map((post) => (
            <Post
                key={post.node.id}
                post={post}
                handleDelete={(id) => handleDelete(id, delete_post)}
            />
        ));
    };

    const render_pagination = (fetchMore, cursor) => {
        return (
            <div className="col s12 center pagination">
                <a className="btn btn-large" onClick={() => handleMore(fetchMore, cursor)}>
                    Show more
                </a>
            </div>
        );
    };

    const handleDelete = (postId, delete_post) => {
        const confirmDelete = confirm('This post will be permanently deleted');
        if (confirmDelete == true) {
            delete_post({
                variables: {
                    postId
                }
            });
        }
    };

    const handleMore = (fetchMore, cursor) => {
        return fetchMore({
            variables: {
                after: cursor
            },
            updateQuery: (previousResult, { fetchMoreResult }) => {
                const newEdges = fetchMoreResult.user.posts.edges;
                const pageInfo = fetchMoreResult.user.posts.pageInfo;

                return newEdges.length
                    ? {
                        user: {
                            id: fetchMoreResult.user.id,
                            posts: {
                                __typename: previousResult.user.posts.__typename,
                                edges: [...previousResult.user.posts.edges, ...newEdges],
                                pageInfo
                            },
                            __typename: previousResult.user.__typename
                        }
                    }
                    : previousResult;
            }
        });
    };

    const deleteError = () => createToast('Error deleting post');

    return (
        <Query query={query} variables={{ first: 5 }}>
            {({ data, error, fetchMore }) => {
                if (error) return <Error render={<p>There was an error fetching post</p>} />;

                const edges = data.user ? data.user.posts.edges : [];
                const hasMore = data.user ? data.user.posts.pageInfo.hasNextPage : false;
                const endCursor = data.user ? data.user.posts.pageInfo.endCursor : '';

                return (
                    <Mutation mutation={deleteMutation} onError={deleteError} 
                        onCompleted={() => client.resetStore()}>
                        {(delete_post) => {
                            return (
                                <div className="main admin-posts">
                                    {edges && (
                                        <div>
                                            {render_page_header()}
                                            <div className="posts-list">
                                                {edges.length > 0
                                                    ? render_posts(edges, delete_post)
                                                    : render_no_post()}
                                            </div>
                                            {hasMore && render_pagination(fetchMore, endCursor)}
                                        </div>
                                    )}
                                </div>
                            );
                        }}
                    </Mutation>
                );
            }}
        </Query>
    );
};

PageView.propTypes = {
    client: types.object.isRequired
};

export default withApollo(PageView);
