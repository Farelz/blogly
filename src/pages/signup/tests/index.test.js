import React from 'react';
import SignUp from '../index';
import { PageView } from '../view';
import { shallow } from 'enzyme';

describe('signup tests', () => {
    it('renders', () => {
        const wrapper = shallow(<SignUp />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders view', () => {
        const props = {
            isLoggingIn: false,
            register_user: jest.fn(),
            fetch_data: jest.fn()
        };

        const view = shallow(<PageView {...props} />);
        expect(view).toMatchSnapshot();
    });
});
