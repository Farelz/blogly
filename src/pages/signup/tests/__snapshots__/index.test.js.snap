// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup tests renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`signup tests renders view 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageView
    fetch_data={[MockFunction]}
    isLoggingIn={false}
    register_user={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="signup-section section center z-depth-1"
        >
          <div
            className="heading"
          >
            <h5>
              Join ReactPress
            </h5>
            <p>
              Sign up to share your story with the world, appreciate stories you love, and more.
            </p>
            <p>
              * Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long
            </p>
          </div>
          <div
            className="login-form"
          >
            <SignupForm
              auth_message=""
              button_class="btn disabled"
              emailValue=""
              input_class=" invalid"
              loggingIn={false}
              nameValue=""
              onChange={[Function]}
              onSubmit={[Function]}
              passwordValue=""
            />
          </div>
          <div
            className="extra-info"
          >
            <p>
              Have an account?
              <Link
                href="/login"
              >
                <a>
                   Login
                </a>
              </Link>
            </p>
          </div>
        </div>
      </div>,
      "className": "auth auth-signup",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="signup-section section center z-depth-1"
        >
          <div
            className="heading"
          >
            <h5>
              Join ReactPress
            </h5>
            <p>
              Sign up to share your story with the world, appreciate stories you love, and more.
            </p>
            <p>
              * Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long
            </p>
          </div>
          <div
            className="login-form"
          >
            <SignupForm
              auth_message=""
              button_class="btn disabled"
              emailValue=""
              input_class=" invalid"
              loggingIn={false}
              nameValue=""
              onChange={[Function]}
              onSubmit={[Function]}
              passwordValue=""
            />
          </div>
          <div
            className="extra-info"
          >
            <p>
              Have an account?
              <Link
                href="/login"
              >
                <a>
                   Login
                </a>
              </Link>
            </p>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="heading"
            >
              <h5>
                Join ReactPress
              </h5>
              <p>
                Sign up to share your story with the world, appreciate stories you love, and more.
              </p>
              <p>
                * Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long
              </p>
            </div>,
            <div
              className="login-form"
            >
              <SignupForm
                auth_message=""
                button_class="btn disabled"
                emailValue=""
                input_class=" invalid"
                loggingIn={false}
                nameValue=""
                onChange={[Function]}
                onSubmit={[Function]}
                passwordValue=""
              />
            </div>,
            <div
              className="extra-info"
            >
              <p>
                Have an account?
                <Link
                  href="/login"
                >
                  <a>
                     Login
                  </a>
                </Link>
              </p>
            </div>,
          ],
          "className": "signup-section section center z-depth-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5>
                  Join ReactPress
                </h5>,
                <p>
                  Sign up to share your story with the world, appreciate stories you love, and more.
                </p>,
                <p>
                  * Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long
                </p>,
              ],
              "className": "heading",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Join ReactPress",
                },
                "ref": null,
                "rendered": "Join ReactPress",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign up to share your story with the world, appreciate stories you love, and more.",
                },
                "ref": null,
                "rendered": "Sign up to share your story with the world, appreciate stories you love, and more.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "* Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long",
                },
                "ref": null,
                "rendered": "* Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <SignupForm
                auth_message=""
                button_class="btn disabled"
                emailValue=""
                input_class=" invalid"
                loggingIn={false}
                nameValue=""
                onChange={[Function]}
                onSubmit={[Function]}
                passwordValue=""
              />,
              "className": "login-form",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "auth_message": "",
                "button_class": "btn disabled",
                "emailValue": "",
                "input_class": " invalid",
                "loggingIn": false,
                "nameValue": "",
                "onChange": [Function],
                "onSubmit": [Function],
                "passwordValue": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Have an account?
                <Link
                  href="/login"
                >
                  <a>
                     Login
                  </a>
                </Link>
              </p>,
              "className": "extra-info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Have an account?",
                  <Link
                    href="/login"
                  >
                    <a>
                       Login
                    </a>
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Have an account?",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <a>
                       Login
                    </a>,
                    "href": "/login",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Login",
                    },
                    "ref": null,
                    "rendered": " Login",
                    "type": "a",
                  },
                  "type": [Function],
                },
              ],
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="signup-section section center z-depth-1"
          >
            <div
              className="heading"
            >
              <h5>
                Join ReactPress
              </h5>
              <p>
                Sign up to share your story with the world, appreciate stories you love, and more.
              </p>
              <p>
                * Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long
              </p>
            </div>
            <div
              className="login-form"
            >
              <SignupForm
                auth_message=""
                button_class="btn disabled"
                emailValue=""
                input_class=" invalid"
                loggingIn={false}
                nameValue=""
                onChange={[Function]}
                onSubmit={[Function]}
                passwordValue=""
              />
            </div>
            <div
              className="extra-info"
            >
              <p>
                Have an account?
                <Link
                  href="/login"
                >
                  <a>
                     Login
                  </a>
                </Link>
              </p>
            </div>
          </div>
        </div>,
        "className": "auth auth-signup",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="signup-section section center z-depth-1"
          >
            <div
              className="heading"
            >
              <h5>
                Join ReactPress
              </h5>
              <p>
                Sign up to share your story with the world, appreciate stories you love, and more.
              </p>
              <p>
                * Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long
              </p>
            </div>
            <div
              className="login-form"
            >
              <SignupForm
                auth_message=""
                button_class="btn disabled"
                emailValue=""
                input_class=" invalid"
                loggingIn={false}
                nameValue=""
                onChange={[Function]}
                onSubmit={[Function]}
                passwordValue=""
              />
            </div>
            <div
              className="extra-info"
            >
              <p>
                Have an account?
                <Link
                  href="/login"
                >
                  <a>
                     Login
                  </a>
                </Link>
              </p>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="heading"
              >
                <h5>
                  Join ReactPress
                </h5>
                <p>
                  Sign up to share your story with the world, appreciate stories you love, and more.
                </p>
                <p>
                  * Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long
                </p>
              </div>,
              <div
                className="login-form"
              >
                <SignupForm
                  auth_message=""
                  button_class="btn disabled"
                  emailValue=""
                  input_class=" invalid"
                  loggingIn={false}
                  nameValue=""
                  onChange={[Function]}
                  onSubmit={[Function]}
                  passwordValue=""
                />
              </div>,
              <div
                className="extra-info"
              >
                <p>
                  Have an account?
                  <Link
                    href="/login"
                  >
                    <a>
                       Login
                    </a>
                  </Link>
                </p>
              </div>,
            ],
            "className": "signup-section section center z-depth-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h5>
                    Join ReactPress
                  </h5>,
                  <p>
                    Sign up to share your story with the world, appreciate stories you love, and more.
                  </p>,
                  <p>
                    * Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long
                  </p>,
                ],
                "className": "heading",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Join ReactPress",
                  },
                  "ref": null,
                  "rendered": "Join ReactPress",
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign up to share your story with the world, appreciate stories you love, and more.",
                  },
                  "ref": null,
                  "rendered": "Sign up to share your story with the world, appreciate stories you love, and more.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "* Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long",
                  },
                  "ref": null,
                  "rendered": "* Password must contain atleast one uppercase letter or one number. And must be atleast 6 characters long",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <SignupForm
                  auth_message=""
                  button_class="btn disabled"
                  emailValue=""
                  input_class=" invalid"
                  loggingIn={false}
                  nameValue=""
                  onChange={[Function]}
                  onSubmit={[Function]}
                  passwordValue=""
                />,
                "className": "login-form",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "auth_message": "",
                  "button_class": "btn disabled",
                  "emailValue": "",
                  "input_class": " invalid",
                  "loggingIn": false,
                  "nameValue": "",
                  "onChange": [Function],
                  "onSubmit": [Function],
                  "passwordValue": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Have an account?
                  <Link
                    href="/login"
                  >
                    <a>
                       Login
                    </a>
                  </Link>
                </p>,
                "className": "extra-info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Have an account?",
                    <Link
                      href="/login"
                    >
                      <a>
                         Login
                      </a>
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Have an account?",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <a>
                         Login
                      </a>,
                      "href": "/login",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": " Login",
                      },
                      "ref": null,
                      "rendered": " Login",
                      "type": "a",
                    },
                    "type": [Function],
                  },
                ],
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
